### PR TITLE
analyze weekly should run weekly

### PR DESCRIPTION
**Pull request by kristapratico**
Per @weshaggard's comment here: https://github.com/Azure/azure-sdk-for-python/pull/33151#discussion_r1393423305 let's only run this analyze stage weekly

### CODE REVIEW

**Comment by weshaggard** [comment]
You should probably also remove this redundant condition in the test template at https://github.com/kristapratico/azure-sdk-for-python/blob/main/eng/pipelines/templates/stages/archetype-sdk-tests.yml#L149

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -146,7 +146,6 @@ stages:
               Location: ${{ coalesce(parameters.Location, cloud.value.Location) }}
               Cloud: ${{ cloud.key }}
 
-- ${{ if contains(variables['Build.DefinitionName'], 'tests-weekly')  }}:
   - template: /eng/pipelines/templates/stages/python-analyze-weekly.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:149]: I suspect you might need to decrease the indent here.

**Review by weshaggard — APPROVED** [review]

**Comment by kristapratico** [comment]
@weshaggard quick question - by adding this condition to the python-analyze-weekly.yml, I was hoping that would mean that this would only run for `- test-weekly` pipelines and not run at all for `- tests`. However, it looks like for services that will just run analyze (e.g. [anomalydetector](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/anomalydetector/tests.yml)), the `- tests` is going to fail:

image

I really only want `-test-weekly`, but I think the automation will set up `- tests` pipelines that will continuously fail with these yml errors. Is there a way around this?

**Comment by benbp** [comment]
@kristapratico in this case I would say let's have the `- tests` pipeline get auto-created, then we can go and manually change it's state to disabled.

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml (hunk 1 above)
Comment (kristapratico) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml]: Thanks. Fixed indentation and tested on a few pipelines.

**Pull request merged**